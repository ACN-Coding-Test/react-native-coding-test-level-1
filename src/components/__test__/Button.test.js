import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';

import {Button} from '..';
import addSeconds from 'date-fns/addSeconds';

describe('testing custom button', () => {
  const wrapper = renderer.create(
    <Button
      onPress={() => console.log('a')}
      backgroundColor="red"
      label="NEXT"
    />,
  );

  it('Should render Button', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should properly render the label', () => {
    const rendered = render(<Button label={'New Label'} />);
    const buttoncomponent = rendered.getByTestId('label');

    expect(buttoncomponent.props.children).toEqual('New Label');
  });

  it('should properly render background red', () => {
    const rendered = render(<Button label={'New Label'} />);
    const buttoncomponent = rendered.getByTestId('button');

    expect(buttoncomponent.props.style).toMatchObject({width: '90%'});
  });

  it('should trigger onPress', () => {
    const onPressMock = jest.fn();

    const rendered = render(
      <Button label={'New Label'} onPress={onPressMock} />,
    );
    const buttoncomponent = rendered.getByTestId('button');

    fireEvent.press(buttoncomponent);

    expect(onPressMock).toBeTruthy();
  });
});
