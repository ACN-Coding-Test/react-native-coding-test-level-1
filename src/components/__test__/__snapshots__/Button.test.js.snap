// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing custom button Should render Button 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#ffffff",
      "borderRadius": 5,
      "elevation": 5,
      "height": 60,
      "marginTop": "5%",
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 5,
      "width": "90%",
    }
  }
  testID="button"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 5,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="touch"
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
                "letterSpacing": 0.5,
                "lineHeight": 24,
                "marginVertical": 2,
              },
              {
                "color": "white",
                "fontSize": 18,
                "letterSpacing": 1.5,
              },
            ],
          ],
        ]
      }
      testID="label"
    >
      NEXT
    </Text>
  </View>
</View>
`;
